stripe/src/main/java/com/stripe/android/paymentsheet/model/ViewState.kt

```diff
@@ -36,9 +43,16 @@ internal sealed class ViewState {
 
         object StartProcessing : PaymentOptions()
 
-        data class FinishProcessing(
+        class FinishProcessing(
             val onComplete: () -> Unit
-        ) : PaymentOptions()
+        ) : PaymentOptions() {
+            /**
+             * Do not compare [onComplete] as part of [equals].
+             */
+            override fun equals(other: Any?): Boolean = other is FinishProcessing
+
+            override fun hashCode(): Int = onComplete.hashCode()
```

Comment:
I think this breaks the hashCode/equals definition. hashCode should also be dependent on the class type only.
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-any/hash-code.html

Reply:
Good point. I suppose we should just not implement `hashCode()`